packer/plugin-getter/plugins.go

```diff
@@ -0,0 +1,421 @@
+package plugingetter
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/hashicorp/go-version"
+	"github.com/hashicorp/packer-plugin-sdk/tmp"
+	"github.com/hashicorp/packer/hcl2template/addrs"
+)
+
+type Requirements []*Requirement
+
+// Requirement describes a required plugin and how it is installed. Usually a list
+// of required plugins is generated from a config file. From it we check what
+// is actually installed and what needs to happen to get in the desired state.
+type Requirement struct {
+	// How the user named the plugin
+	Accessor string
+
+	// Something like github.com/hashicorp/packer-plugin-amazon
```

Comment:
we're probably gonna have people trying to reach plugins on their local filesystem

Reply:
Yup ! This should still be okay, packer will ask for a plugin named `github.com/hashicorp/packer-plugin-amazon` and a local file system plugin-getter should be able to know what to do with that. 🙂